File: examples/python/gi/list-connections.py
#!/usr/bin/env python
# SPDX-License-Identifier: GPL-2.0-or-later
#
# Copyright (C) 2012 - 2014 Red Hat, Inc.
#

import gi

gi.require_version("NM", "1.0")
from gi.repository import NM

# This example asks settings service for all configured connections.


def print_values(setting, key, value, flags, data):
    print("  %s.%s: %s" % (setting.get_name(), key, value))


if __name__ == "__main__":
    # create Client object
    client = NM.Client.new(None)

    # get all connections
    connections = client.get_connections()

    # print the connections' details
    for c in connections:
        print("=== %s : %s ===" % (c.get_id(), c.get_path()))
        c.for_each_setting_value(print_values, None)
        print("\n")

File: examples/lua/lgi/list-connections.lua
#!/usr/bin/env lua
-- SPDX-License-Identifier: GPL-2.0+
--
-- Copyright (C) 2015 Red Hat, Inc.
--

-- Getting and printing all NetworkManager connection in Lua.
-- The example uses libnm library using GObject introspection via Lua lgi module.
-- Most distribution ship the module as lua-lgi package.
-- libnm guide:   https://developer.gnome.org/libnm/1.0/
-- Lua-lgi guide: https://github.com/pavouk/lgi/blob/master/docs/guide.md

local lgi = require 'lgi'
local GLib = lgi.GLib
local NM = lgi.NM

function print_values(setting, key, value, flags, data)
  print(string.format("  %s.%s: %s", setting:get_name(), key, value and tostring(value.value) or "nil"))
end


---------------------------
-- Main code starts here --
---------------------------
-- parse command-line arguments
local details = ...
if (details and not string.find("details", details, 1)) then
  print(string.format("Usage: %s [details]", arg[0]:gsub(".*/","")))
    os.exit(1)
end

-- create Client object
local client = NM.Client.new()

-- get connections
connections = client:get_connections()

-- print the connections' details
print(string.format("There are %s connection profiles.", #connections))
for i, c in pairs(connections) do
  print(string.format("=== %s | %25s | %s ", c:get_uuid(), c:get_id(), c:get_path()))
  if details then
    c:for_each_setting_value(print_values, nil)
    print("\n")
  end
end

